coderd/database/migrations/000168_pg_coord_tailnet_v2_api.up.sql

```diff
@@ -0,0 +1,72 @@
+BEGIN;
+
+CREATE TYPE tailnet_status AS ENUM (
+	'ok',
+	'lost'
+);
+
+CREATE TABLE tailnet_peers (
+	id uuid NOT NULL,
+	coordinator_id uuid NOT NULL,
+	updated_at timestamp with time zone NOT NULL,
+	node bytea NOT NULL,
+	status tailnet_status DEFAULT 'ok'::tailnet_status NOT NULL,
+	PRIMARY KEY (id, coordinator_id),
+	FOREIGN KEY (coordinator_id) REFERENCES tailnet_coordinators(id) ON DELETE CASCADE
+);
+
+-- For shutting down / GC a coordinator
+CREATE INDEX idx_tailnet_peers_coordinator ON tailnet_peers (coordinator_id);
+
+-- Any time tailnet_peers table changes, send an update with the affected peer ID.
+CREATE FUNCTION tailnet_notify_peer_change() RETURNS trigger
+	LANGUAGE plpgsql
+AS $$
+BEGIN
+	IF (OLD IS NOT NULL) THEN
+		PERFORM pg_notify('tailnet_peer_update', OLD.id::text);
+		RETURN NULL;
+	END IF;
+	IF (NEW IS NOT NULL) THEN
+		PERFORM pg_notify('tailnet_peer_update', NEW.id::text);
+		RETURN NULL;
+	END IF;
+END;
+$$;
+
+CREATE TRIGGER tailnet_notify_peer_change
+	AFTER INSERT OR UPDATE OR DELETE ON tailnet_peers
+	FOR EACH ROW
+EXECUTE PROCEDURE tailnet_notify_peer_change();
+
+CREATE TABLE tailnet_tunnels (
+	coordinator_id uuid NOT NULL,
+	-- we don't keep foreign keys for src_id and dst_id because the coordinator doesn't
+	-- strictly order creating the peers and creating the tunnels
```

Comment:
Why don't we always create the peers first before tunnels?

Reply:
Adding peers occurs when we have the Node object for that peer.

The tunnel is added right away for CLI clients, and allows it to get the Agent node.  You want this to happen ASAP, rather than gating it behind the Client figuring out it's own node.

These things happen async in the PG Coordinator implementation, so having an ordering dependency in the DB complicates things significantly.

Reply:
If we're adding peers only when we get the node object, why don't we just insert a peer with a NULL node object so it's easier to track and we avoid having tunnels that may or may not reference real peers?

Reply:
Can you be more concrete about the benefit you see re: being "easier to track"?

In terms of getting tunnels that don't reference peers, I plan to use SQL joins when figuring out what nodes each side of the tunnel needs, so if the corresponding peer is missing, it just won't return any rows for that tunnel.

In the PG Coordinator implementation, it's separate subcomponents that handle updating peers and tunnels, and so blocking one type of update contingent on another is a synchronization problem that adds complexity.

Reply:
If you already have a plan for dealing with tunnels that don't have matching peers then I think this is fine.